[docs] Release notes for 2.23.0.0-b710 (#23687)
yugabyte/yugabyte-db: docs/content/preview/releases/ybdb-releases/v2.23.md

* Addresses a race condition in dynamic table creation, enhancing stability during table and tablet initialization. {{<issue 22408>}}
* Refines the logic to remove `BEGIN` record when no DML records are added, preventing potential virtual WAL crashes. {{<issue 21646>}}
* Resolves "could not open relation" error by updating slot creation method and simplifying yb_read_time logic. {{<issue 22398>}}
* Enables support for dynamically allotted oids data types in CDC to prevent system crashes. {{<issue 23179>}}
* Allows handling of non-eligible table cleanup in CDC stream loading even after table drop, preventing master crash. {{<issue 23278>}}
* Reduces total inserts from 5k/thread to 2.5k/thread for clearer consumption of expected records. {{<issue 23128>}}
* Prevents failures in decoding change events by refreshing `cached_schema_details` when executing a new `GetChanges` request if the client indicates a necessity for the schema. {{<issue 20698>}}
* Allows pg_replication_slots to return an empty response instead of an error when `ysql_yb_enable_replication_commands` GFlag is false. {{<issue 23096>}}

> <details>

Reviewer: what's this

Author: typo - it should be </details>. Thanks

For an RSS feed of all release series, point your feed reader to the [RSS feed for releases](../../index.xml).

## v2.23.0.0 - September 6, 2024 {#v2.23.0.0}

**Build:** `2.23.0.0-b710`

### Downloads

<ul class="nav yb-pills">
  <li>
    <a href="https://downloads.yugabyte.com/releases/2.23.0.0/yugabyte-2.23.0.0-b710-darwin-x86_64.tar.gz">
      <i class="fa-brands fa-apple"></i>
      <span>macOS</span>
    </a>
  </li>
  <li>
    <a href="https://downloads.yugabyte.com/releases/2.23.0.0/yugabyte-2.23.0.0-b710-linux-x86_64.tar.gz">
      <i class="fa-brands fa-linux"></i>
      <span>Linux x86</span>
    </a>
  </li>
  <li>
    <a href="https://downloads.yugabyte.com/releases/2.23.0.0/yugabyte-2.23.0.0-b710-el8-aarch64.tar.gz">
      <i class="fa-brands fa-linux"></i>
      <span>Linux ARM</span>
    </a>
  </li>
</ul>

**Docker:**

```sh
docker pull yugabytedb/yugabyte:2.23.0.0-b710
```

### Highlights

We're excited to announce the [technical preview](/preview/releases/versioning/#tech-preview-tp) of four powerful new features in YugabyteDB aimed at simplifying operations and enhancing functionality.

> **Database instant cloning** {{<badge/tp>}}

Reviewer: ```suggestion
**Instant database cloning** {{<badge/tp>}}
```

> [Database Cloning](/preview/manage/backup-restore/lightweight-db-clone/) allows you to quickly create independent copies of your database for data recovery, development, and testing. Clones can be created in seconds, and initially consume no additional disk space, because they share the same data files as the original database at creation but operate independently thereafter. Clones can be created as of now, or as of any time in the recent past (within a configurable history retention period), enabling developers to experiment without impacting production performance. This feature also provides a safety net for rapid recovery from accidental data loss or corruption.

Reviewer: ```suggestion
[Database Cloning](/preview/manage/backup-restore/lightweight-db-clone/) allows you to quickly create independent copies of your database for data recovery, development, and testing. Clones can be created in seconds, and initially consume no additional disk space because they share the same data files as the original database. Clones can be created as of now, or as of any time in the recent past (within a configurable history retention period), so developers can experiment without impacting production performance. Clones also provide a safety net for rapid recovery from accidental data loss or corruption.
```

**pg_cron extension** {{<badge/tp>}}

We're introducing support for the [pg_cron extension](/preview/explore/ysql-language-features/pg-extensions/extension-pgcron/), which brings a cron-based job scheduler directly into the database. With pg_cron, you can schedule YSQL commands using familiar cron syntax, including jobs on intervals as fine as seconds. While pg_cron operates on a single node known as the pg_cron leader, the queries it schedules leverage the full resources of the distributed cluster. In case of node failure, leadership is automatically transferred, ensuring continuous availability and smooth scheduling operations.

> **DB Scoped xCluster replication** {{<badge/tp>}}

Reviewer: ```suggestion
**Semi-automatic xCluster replication** {{<badge/tp>}}
```

> [DB Scoped xCluster replication](/preview/deploy/multi-dc/async-replication/async-replication-transactional/) simplifies the management of YSQL transactional xCluster replication by operating at the database level instead of the table level. This reduces the need for IT/admin involvement when tables are created or dropped, as xCluster management is only required for adding or removing entire databases from replication. DDL operations can be performed by authorized users or DBAs without needing elevated privileges. DB Scoped xCluster supports YSQL transactional replication and is ideal for Disaster Recovery (DR) purposes.

Reviewer: ```suggestion
[Semi-automatic xCluster replication](/preview/deploy/multi-dc/async-replication/async-replication-transactional/) simplifies the management of YSQL transactional xCluster replication by operating at the database level instead of the table level. This reduces the need for IT admin involvement when tables are created or dropped, as xCluster management is only required for adding or removing entire databases from replication. DDL operations can be performed by authorized users or DBAs without needing elevated privileges. Semi-automatic xCluster supports YSQL transactional replication and is ideal for Disaster Recovery (DR) purposes.
```

> **Improvement to Backward scans** {{<badge/tp>}}

Reviewer: ```suggestion
**Improvement to backward scans** {{<badge/tp>}}
```

> We have improved the performance of backward scans by 10x. For example, descending scans on tables that are stored in ascending order are now much faster. Previously, backward scans were significantly slower than forward scans in Yugabyte, causing the query optimizer to implement various workarounds including, adding a sort operator on top of the forward scan operator to keep the backward scan pattern comparable. Such optimizer tricks might have been ineffective in complex query plans. The improvements to backward scans performance now allow such queries to be faster out of the box. The feature can be enabled by using the preview flag, `use_fast_backward_scan`.

Reviewer: ```suggestion
We've improved the performance of backward scans by 10x. For example, descending scans on tables that are stored in ascending order are now much faster. Previously, backward scans were significantly slower than forward scans in YugabyteDB, causing the query optimizer to implement various workarounds, including adding a sort operator on top of the forward scan operator to keep the backward scan pattern comparable. Such optimizer tricks might have been ineffective in complex query plans. The improvements to backward scan performance now allows such queries to be faster out of the box. You enable the feature using the `use_fast_backward_scan` preview flag.
```
